dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataLogsJdbcTask.java

```diff
@@ -299,7 +308,12 @@ private Boolean isValid(@Nonnull JdbcTemplate template, @Nonnull String expressi
   }
 
   @Override
-  public String toString() {
-    return getSql(Predicates.alwaysTrue());
+  public String describeSourceData() {
+    return sql.map(AbstractTask::createSourceDataDescriptionForQuery)
```

Comment:
What is a downside of running `getSql` here? This would simplify `describeSourceData` function as we do not need to handle empty Optional.

Reply:
The current implementation doesn't allow running `getSql` here easily, because we do not have the right `Predicate`. The `alwaysTrue` predicate was used before, which led to the confusion that I've described in the PR description - the printed query and the executed query were different, which is undesirable.

The idea of passing the `Predicate` into the `getSql` function itself is a bit convoluted. It would be better to prepare the SQL query first, save it in the `*Task` class and then execute the task. However, the check for the existing columns is being performed in the same stage as executing the query (right before the query is executed), so there is no separation of creating a query and then creating a task with this query.

Because such an SQL query is dynamically generated based on the schema of the database that is known only after connecting to the database, I rely on the fact that the query is executed first and printed only later in the summary at the end of dumper execution. Therefore saving it in the field in the `*Task` class and retrieving it here makes sense.

For the long-term: A better solution would require changing the whole structure of how tasks are prepared, so that the SQL query could be passed to a `*Task` class and be immutable there, but this would be a much bigger change.